### DISCUSSION TITLE
Failed to set texture to a part of a 3D glb model,the texture malposition

### DISCUSSION THREAD

**Discussion by whereisnothing** [google/filament#5919] [General]
**Describe the bug**
Failed to set texture to a part of a 3D glb model,the position of the texture is wrong.See screenshot.

**To Reproduce**
When I use the next lines of code to set a new texture to a part of a 3D glb model,it doesn't succeed,that is: the new texture malposition,it can't match its original image(right eye image,see screenshot).Please help me out,appreciate it.

```kotlin
    fun changeTexture() {
        val rm = engine.renderableManager
        for (entity in asset!!.renderableEntities) {
            val renderable = rm.getInstance(entity)
            if (renderable == 0) {
                continue
            }
            val partName = asset!!.getName(entity);
            if (partName == "eye") {
                val mi = rm.getMaterialInstanceAt(renderable, 0)
                val baseColor = loadTexture(
                    engine,
                    surfaceView!!.resources,
                    R.drawable.new_eye,
                    TextureType.COLOR
                )
                val sampler = TextureSampler()
                sampler.anisotropy = 8.0f
                sampler.setWrapModeR(TextureSampler.WrapMode.REPEAT);
                sampler.setWrapModeS(TextureSampler.WrapMode.REPEAT);
                sampler.setWrapModeT(TextureSampler.WrapMode.REPEAT);
                mi.setParameter("baseColorIndex", 0)
                mi.setParameter("baseColorMap", baseColor, sampler)
            }
        }
    }
```

**Expected behavior**
The changed texture should take effect.

**Screenshots**
![head](https://user-images.githubusercontent.com/6499374/185359065-b29787ef-dba3-49a5-974d-ea98b8576704.jpg)

![jichu_eye](https://user-images.githubusercontent.com/6499374/184475158-dbc77cfb-c1ca-48a0-8350-0705bdf13171.png)

**Logs**
If applicable, copy logs from your console here. Please *do not*
use screenshots of logs, copy them as text.

**Desktop (please complete the following information):**
 - GPU: [e.g. NVIDIA GTX 1080]
 - Backend: [OpenGL/Vulkan]

**Smartphone (please complete the following information):**
 - Device: Huawei
 - OS: Android 10

**Additional context**

**Comment by prideout** [comment]
Does the original glb have a UV matrix?  Does the original glb have an eye texture at all?  It looks like you're using a texture atlas for the eyes, so you might need to adjust `baseColorUvMatrix` so that it translates the UVs appropriately.  This would be true with any renderer, not just filament.

**Reply by whereisnothing** [reply]
The original glb have an eye texture ,but I hide it.Then I use the second image in the screenshot list to change the eye texture of the glb model.The second image shows three states of the eye:Fully open eye,semiopen eye,closed eye.If I choose to adjust `baseColorUvMatrix` ,does it mean I need to get the `baseColorUvMatrix` value from UI designers?
